src/lib/components/Modal/Modal.jsx

```diff
@@ -78,54 +79,120 @@ export const Modal = ({
   autoFocus,
   children,
   closeButtonRef,
+  focusEnforce,
   portalId,
   position,
+  preventScrollUnderneath,
   primaryButtonRef,
   size,
   ...restProps
 }) => {
   const childrenWrapperRef = useRef();
 
-  const keyPressHandler = (e) => {
-    if (e.key === 'Escape' && closeButtonRef?.current != null) {
-      closeButtonRef.current.click();
+  useEffect(() => {
+    if (!autoFocus) {
+      return () => {};
     }
 
-    if (e.key === 'Enter' && e.target.nodeName !== 'BUTTON' && primaryButtonRef?.current != null) {
-      primaryButtonRef.current.click();
+    // If `autoFocus` is set to `true`, following code finds all focusable elements and among them
```

Comment:
> following code

How about moving this into a function with a self-explanatory name like `focusFirstFocusableElement(autoFocus, enableFocusTrap)`? Or maybe even external helpers? The code here starts to get a little bit long and more difficult to read.

Let @bedrich-schindler or @mbohal decide, it's just my personal feeling.

Reply:
Why not. I just want to mention that if we want to make it nicer, I would refactor whole component. One small extraction to the helper wouldn't help so much and it would be even more messy because nobody would know "the style in which is component implemented".

Reply:
Tell me whether you will refactor it or not. If yes, I will review it after the component is refactored.

Reply:
Should be ready. @bedrich-schindler